ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryBuilder.java

```diff
@@ -528,8 +529,14 @@ public static Map<String, Object> buildStreamsProperties(
       final KsqlConfig config,
       final ProcessingLogContext processingLogContext
   ) {
-    final Map<String, Object> newStreamsProperties
-        = new HashMap<>(config.getKsqlStreamConfigProps(applicationId));
+    Map<String, Object> newStreamsProperties = new HashMap<>(config.getKsqlStreamConfigProps(applicationId));
+
+    if (!KsqlConfig.DeploymentType.confluent.toString().equals(config.getString(KSQL_DEPLOYMENT_TYPE_CONFIG))) {
```

Comment:
Is it worth a comment indicating why we don't want the `addConfluentMetricsContextConfigsKafka` configs included for the `confluent` deployment type?  It's counterintuitive :slightly_smiling_face: